index.js

```diff
@@ -51,23 +51,45 @@ exports.extend = function(app, options) {
     const mimeTypeLimit = options.mimeTypeLimit ? !Array.isArray(options.mimeTypeLimit) ? [options.mimeTypeLimit] : options.mimeTypeLimit : null;
     delete options.restrictMultiple;
     delete options.mimeTypeLimit;
-    
+
     app.use(busboy(options));
 
     app.use((req, res, next) => {
         var allowUpload = true;
-        
+
         req.body = req.body || {};
         req.files = req.files || {};
 
         if (req.is('json') && req.readable) {
-            jsonBody(req, res, options, (err, body) => {
```

Comment:
I might prefer that we just make `jsonBody` configurable if this is really a use case that we want to support.

I have a few concerns:

- We're making the request object that much larger by retaining both the string and parsed body which might increase memory consumption. Not necessarily a deal killer, but it's probably not great to punish people by default.
- By retaining a second reference to the original body, it _may_, at least in our case, retain some references to a potentially insecure request if there is some sanitization that happens during the revival. I don't want to say specifically how we handle security, but retaining the raw body seems to add some potential for bad stuff.

----

That being said, allowing `jsonBody` to be configurable then becomes part of the API and we have to support the API of `jsonBody` forever, though since we already pass `options` into `jsonBody` you can make the argument that they're already intertwined.

I think a simple

```js
exports.extend = function(app, options) {
  // ...
  const jsonBodyFn = options.jsonBody || jsonBody;
  // ...
  if (req.is('json') && req.readable) {
    jsonBody(req, res, options, (err, body) => {
      // ..
    }
    // ..
  }
```

might be a nice compromise here